### ISSUE TITLE
React.js warning about handlers returning false

### ISSUE THREAD

**Issue by skrat** [reagent-project/reagent#78]
This is new to 0.12.1 I guess. There are many cases where the event handler might incidentaly return false, and handling by hand to rid the warning is annoying and pollutes the code. Eg.

```
[:button {:on-click #(swap! foo not)}] ;; foo being an atom holding a boolean value
```

I propose returning null, true or undefined by reagent, ie. wrapping the handlers in

`(fn [& args] (do (apply handler args) nil))`

**Comment by cigitia** [comment]
A note for any future readers: the warning also occurs if the return value is the boolean `true`, not only the boolean `false`, which confused me for a while. Returning `nil` does not trigger the warning.

**Comment by danschumann** [comment]
It's impossible to find what is actually calling this.  I have one button that returns true ( now ).  The problem is coffeescript returns everything, and the stack trace (if I edit console.warn to console.trace [ which works for other warnings ] ), Does not escape react into my application code, so I literally have no clue what is firing this, even if I cause it to throw an error instead of warning in react.js:21560 

Is there any way (even with chromes async checkbox [ which is on ] ), so get a stack trace all the way up to the offending callback?

**Comment by danschumann** [comment]
It turns out it is offended when you return true as well ( trying to please it ).

I was able to `console.log` line 11367 (react-with-addons), inside of dispatchEvent, above `     ReactUpdates.batchedUpdates(handleTopLevelImpl, bookKeeping);`, which, looking at arguments, I noticed the event was indeed the button I told to return true ( to please react ), but it gives a misleading argument about returning `false`, even when you return `true`.  I'll be glad when this notice is just gone.

**Comment by city41** [comment]
With CoffeeScript you might have to throw in a `return;` to get it to just return undefined.

**Comment by jw-00000** [comment]
For future readers: React v0.13 has the same behavior as v0.12 — unlike what was said before in this conversation, and unlike what React's documentation said. I've updated the [wiki page](https://github.com/reagent-project/reagent/wiki/Beware-Event-Handlers-Returning-False).

Relevant code in React [v0.11](https://github.com/facebook/react/blob/0.11-stable/src/browser/eventPlugins/SimpleEventPlugin.js#L310-L313), [v0.12](https://github.com/facebook/react/blob/0.12-stable/src/browser/eventPlugins/SimpleEventPlugin.js#L314-L317), [v0.13](https://github.com/facebook/react/blob/0.12-stable/src/browser/eventPlugins/SimpleEventPlugin.js#L314-L317).

To summarize:

* v0.11, v0.12, and v0.13 stop event propagation on returning false. (Only false, not falsy values.)
* v0.12 and v0.13 issue a deprecation warning if a boolean was returned.

Looking at React's master branch, it seems like this code has undergone a rewrite, so I guess that in v0.14 the return value _will_ be ignored.
